packages/components/paginator/src/paginator.spec.ts

```diff
@@ -1,616 +1,383 @@
+    it('should have a enabled next button', () => {
+      const button = el.renderRoot.querySelector(':nth-child(2 of sl-button.nav)');
 
-      expect(el.page).to.equal(1);
+      expect(button).to.exist;
+      expect(button).to.have.attribute('aria-label', 'Go to the next page (2)');
+      expect(button).not.to.match(':disabled');
     });
 
-    it('should have previous and next buttons with proper aria-labels', async () => {
-      el.page = 3;
+    it('should disable the next button when the current page is the last', async () => {
+      el.page = 19;
       await el.updateComplete;
 
-      const prev = el.renderRoot.querySelector('sl-button.prev') as Button,
-        next = el.renderRoot.querySelector('sl-button.next') as Button;
-
-      expect(prev).to.exist;
-      expect(next).to.exist;
+      const button = el.renderRoot.querySelector(':nth-child(2 of sl-button.nav)');
 
-      const prevAriaLabel = prev.ariaLabel,
-        nextAriaLabel = next.ariaLabel;
-
-      expect(prevAriaLabel).to.equal('Go to the previous page (2)');
-      expect(nextAriaLabel).to.equal('Go to the next page (4)');
+      expect(button).to.exist;
+      expect(button).to.match(':disabled');
     });
 
-    it('should emit an sl-page-change event when the page has changed', async () => {
-      const onPageChange = spy();
-
-      el.addEventListener('sl-page-change', onPageChange);
-
-      el.page = 5;
-      await el.updateComplete;
+    it('should have the current page set to the first page', () => {
+      const button = el.renderRoot.querySelector('sl-button[aria-current="page"]');
 
-      expect(onPageChange).to.have.been.called;
+      expect(button).to.exist;
+      expect(button).to.have.trimmed.text('1');
+      expect(button).to.match('.current');
     });
 
-    it('should send an announcement when the page has changed', async () => {
-      el.page = 5;
-      await el.updateComplete;
-      await new Promise(resolve => setTimeout(resolve, 100));
+    it('should have a page size of 10', () => {
+      const buttons = el.renderRoot.querySelectorAll('sl-button.page'),
+        labels = Array.from(buttons).map(button => button.textContent?.trim());
 
-      expect((sendToAnnouncerSpy.getCall(-1).args[0] as SlAnnounceEvent).detail.message).to.equal('Page 5 of 10');
+      expect(el.pageSize).to.equal(10);
+      expect(buttons).to.have.lengthOf(20);
+      expect(labels).to.deep.equal(Array.from({ length: 20 }).map((_, i) => `${i + 1}`));
     });
 
-    it('should set the first page when items per page amount has changed', async () => {
-      el.pageSize = 10;
-      await el.updateComplete;
-
-      expect(el.page).to.equal(1);
-    });
+    it('should only show 10 pages', () => {
+      const buttons = Array.from(el.renderRoot.querySelectorAll<Button>('sl-button.page')).filter(
+        button => button.style.display !== 'none'
+      );
 
-    it('should set the first page when items total amount of items has changed', async () => {
```

Comment:
We should also check whether it's setting the first page when the total amount of items has changed.

Reply:
Fixed